scripts/exglobal_archive.sh

```diff
@@ -264,13 +264,32 @@ if [[ ${HPSSARCH} = "YES" || ${LOCALARCH} = "YES" ]]; then
         set +e
         ${TARCMD} -P -cvf "${ATARDIR}/${PDY}${cyc}/${targrp}.tar" $(cat "${ARCH_LIST}/${targrp}.txt")
         status=$?
+
+        # Test gdas.tas and gdas_restarta.tar for rstprod and change group if so
         case ${targrp} in
             'gdas'|'gdas_restarta')
-                ${HSICMD} chgrp rstprod "${ATARDIR}/${CDATE}/${targrp}.tar"
-                ${HSICMD} chmod 640 "${ATARDIR}/${CDATE}/${targrp}.tar"
+                has_rstprod="NO"
+                # Test for rstprod in each archived file
+                while IFS= read -r file; do
+                    if [[ -f ${file} ]]; then
+                        group=$( stat -c "%G" "${file}" )
+                        if [[ "${group}" == "rstprod" ]]; then
+                            has_rstprod="YES"
+                            break
+                        fi
+                    fi
+                done < "${ARCH_LIST}/${targrp}.txt"
+
+                # Change group to rstprod if it was found
+                if [[ "${has_rstprod}" == "YES" ]]; then
+                    ${HSICMD} chgrp rstprod "${ATARDIR}/${CDATE}/${targrp}.tar"
+                    ${HSICMD} chmod 640 "${ATARDIR}/${CDATE}/${targrp}.tar"
+                fi
                 ;;
             *) ;;
         esac
+
+        # For safety, test if the htar/tar command failed after changing groups
```

Comment:
`status` is the return code for `${TARCMD}`, so it is not going to check on the state of `chgrp`
Should this next if-block be before the test for containment of rstprod?

Reply:
@aerorahul No, I think it should be after `chgrp`.  That way, if the `htar` command fails due to something like a timeout, the `chgrp` command still executes to protect the contents.  That said, it would probably be a good idea to have a test for `chgrp` as well.  I will add that in.